python/lsst/ts/MTAlignment/alignment_model.py

```diff
@@ -121,12 +118,16 @@ def connected(self):
             or self.writer.is_closing()
         )
 
-    async def disconnect(self):
+    async def disconnect(self) -> None:
         """Disconnect from the T2SA. A no-op if not connected."""
         if self.writer is not None:
+            try:
+                await self.reset_t2sa_simulation_mode()
+            except Exception:
+                self.log.exception("Failed to reset t2sa simulation mode. Ignoring.")
```

Comment:
I gently suggest:
```
writer = self.writer
self.writer = None
if writer is not None:
   ...
```
so that self.writer is always cleared. Using try/finally to set it to None would also be fine.